Non-intrusive fix for Changeset::Pipe and latest transproc

Kukunin opened the pull request.

Kukunin committed:
Fix Changeset::Pipe with latest transproc

Current version of `Changeset::Pipe` stops working with master `transproc` gem
It's because `transproc` implemented `Transproc::Transformer#t` method
for better DSL. Transformer also started to use that `#t` method internally.

Pipe was implemented as `Transproc::Registry`, but after some point
https://github.com/rom-rb/rom-repository/commit/eb9c0100749a9ca032bb61099217b32047acc9cf#diff-111faacc1a95b1332cab55229f2da173R6
it became `Transproc::Transformer` as well as `Transproc::Registry`.

Both `Transproc::Registry` and `Transproc::Transformer` defines `.[]` method
and also `#t` in master `transproc`.
Being using together is not considered valid situation for `transproc`

Those methods conficted and caused the bug

solnic commented on lib/rom/repository/changeset/pipe.rb:
```diff
@@ -5,6 +5,7 @@ module ROM
   class Changeset
     # Built-in ROM::Repository transformations available Changeset
     #
+    # @todo merge into PipeRegistry when transproc <= 0.4.2 support is dropped
```
Can we do that already? If it can simplify rom-repo then I'm all for it. I should be able to wrap up transproc 1.0.0 this way

Kukunin committed:
Add explicit transproc dependency to Gemfile

Due ROM::Changeset uses Transproc directly, not only via ROM::Mapper

Kukunin commented:
@solnic please review. Just reverted two latest commits, also added transproc to the Gemfile. 

I believe transproc should be in rom-repository's Gemfile, because changeset uses it directly.
I saw other dependencies are from master as well, so it seems ok.

Kukunin commented:
Actually, I'm not sure if I had to add `transproc` to the `gemspec`, not Gemfile

solnic commented:
@Kukunin oh that's my bad, we definitely need `transproc` in the gemspec! that's a good catch, thank you
